### ISSUE TITLE
[Bug]: Dropdown quickly dismisses when we press the tab key from keyboard, when viewport is smaller [320X256] 400% zoom.]

### ISSUE THREAD

**Issue by v-poornima** [microsoft/fluentui#24784]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
  OS: Windows 10 19044.1889
 
  Memory: 146 GB used and 329 GB free
Browsers:
  Edge: Version 105.0.1343.27 (Official build) (64-bit)
```

### Are you reporting Accessibility issue?
Yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown 

### Bug Description

**Repro Steps:**
Fluent UI Link : https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown
1.open the above link and move to the **"Dropdown with custom selected option styles"**
2. At 400%, Tab to and "Multi-select controlled example" drop-down
3. Dialog appears for the fruit and vegetables.

Video Link for the Bug:

https://user-images.githubusercontent.com/113108308/190116193-296635de-9136-434f-8b40-d749bbcddea0.mp4

**Note:** The panel opens when viewport is smaller, but dropdown dismisses.

**Issue:** Pressing tab key from keyboard dialog is closing. Tabbing should go to the close button. I think the dialog should have a focus trap as well.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thank you for filing this issue with us and sharing the repro steps. This issue has been queued for upcoming investigations to determine if the behavior is an issue, or a regression, or if it was by design at the time. To set expectations the developers will review this issue once capacity allows and all updates will be posted on the issue accordingly.

cc @smhigley
